contracts/GenericLender/GenericAave.sol

```diff
@@ -0,0 +1,176 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.6.12;
+pragma experimental ABIEncoderV2;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+
+import "./GenericLenderBase.sol";
+import "../Interfaces/Aave/IAToken.sol";
+import "../Interfaces/Aave/ILendingPool.sol";
+import "../Interfaces/Aave/IProtocolDataProvider.sol";
+import "../Interfaces/Aave/IReserveInterestRateStrategy.sol";
+import "../Libraries/Aave/DataTypes.sol";
+
+/********************
+ *   A lender plugin for LenderYieldOptimiser for any erc20 asset on Aave (not eth)
+ *   Made by SamPriestley.com
+ *   https://github.com/Grandthrax/yearnv2/blob/master/contracts/GenericLender/GenericCream.sol
+ *
+ ********************* */
+
+contract GenericAave is GenericLenderBase {
+    using SafeERC20 for IERC20;
+    using Address for address;
+    using SafeMath for uint256;
+    
+    uint256 private constant blocksPerYear = 2_300_000;
+
+    IProtocolDataProvider public immutable protocolDataProvider;
+    IAToken public immutable aToken;
+
+    constructor(
+        address _strategy,
+        string memory name,
+        IProtocolDataProvider _protocolDataProvider,
+        IAToken _aToken
+    ) public GenericLenderBase(_strategy, name) {
+        protocolDataProvider = _protocolDataProvider;
+        aToken = _aToken;
+
+        require(ILendingPool(_protocolDataProvider.ADDRESSES_PROVIDER().getLendingPool()).getReserveData(address(want)).aTokenAddress == address(_aToken), "WRONG ATOKEN");
+
+        want.approve(address(_protocolDataProvider.ADDRESSES_PROVIDER().getLendingPool()), type(uint256).max);
+    }
+
+    function nav() external view override returns (uint256) {
+        return _nav();
+    }
+
+    function _nav() internal view returns (uint256) {
+        return want.balanceOf(address(this)).add(underlyingBalanceStored());
+    }
+
+    function underlyingBalanceStored() public view returns (uint256 balance) {
+        balance = aToken.balanceOf(address(this));
```

Comment:
don't you need to return the balance of the underlying? this is just returning the balance of the aToken which is not the same.

Reply:
see https://github.com/Grandthrax/yearnV2-generic-lender-strat/pull/8#discussion_r591620054